test/compiler/inline.jl

```diff
@@ -319,6 +319,62 @@ function f37555(x::Int; kwargs...)
 end
 @test f37555(1) == 1
 
+# Issue #18773 - Specifying `@noinline` at callsite
+# Ensure `@noinline` in caller overrides `@inline` in callee
+@inline f18773(x) = x
+g18773(x) = @noinline f18773(x)
+let ci = code_typed(g18773, Tuple{Int})[1].first
+    @test length(ci.code) == 2 &&
+        isexpr(ci.code[1], :invoke) &&
+        ci.code[1].args[1].def.name == :f18773
+end
```

Comment:
Let's make this more robust against the future changes in IRs:
```suggestion
let ci = first(@code_typed g18773(1))
    @test any(ci.code) do x
        isexpr(x, :invoke) && x.args[1].def.name === :f18773
    end
end
```

Reply:
When I used the `@code_typed` macro in any tests instead of the code_typed, I get a macro undefined error. Other than that I incorporated these changes